src/cryptoadvance/specter/notifications/websockets_server_client.py

```diff
@@ -0,0 +1,391 @@
+"""
+This file enabled to keep an open websocket connection with the browser sessions.
+"""
+import logging
+from queue import Queue
+
+logger = logging.getLogger(__name__)
+
+
+import secrets
+import asyncio
+import time, json
+import websockets
+import threading
+from ..helpers import robust_json_dumps
+
+
+class WebsocketsBase:
+    """
+    A base class that keeps a websockets connection or server running forever in a different thread
+    """
+
+    def __init__(self, port):
+        self.domain = "localhost"  # the client and server always run on localhost
+        self.port = port
+        self.quit = False
+        self.started = False
+        self.loop = None
+
+    def quit_now(self):
+        self.quit = True
+
+    def forever_function(self):
+        "This is the function that will contain an endless loop"
+        pass
+
+    def _forever_thread(self):
+        loop = asyncio.new_event_loop()
```

Comment:
Why is it necessary to create a new event-loop? Can't we use the exisitng one? like this:
```suggestion
        loop = asyncio.get_event_loop()
```

Reply:
You have to create a new loop for every thread. Otherwise you get `    raise RuntimeError('There is no current event loop in thread %r.'`